Code/Editor/MainWindow.cpp

```diff
@@ -2411,14 +2416,15 @@ void MainWindow::InitializeMenus()
     }
 
     // Add Menus to MenuBar
-    QMenuBar* menuBar = this->menuBar();
-    menuBar->clear();
-    menuBar->addMenu(m_menuManagerInterface->GetMenu(FileMenuIdentifier));
-    menuBar->addMenu(m_menuManagerInterface->GetMenu(EditMenuIdentifier));
-    menuBar->addMenu(m_menuManagerInterface->GetMenu(GameMenuIdentifier));
-    menuBar->addMenu(m_menuManagerInterface->GetMenu(ToolsMenuIdentifier));
-    menuBar->addMenu(m_menuManagerInterface->GetMenu(ViewMenuIdentifier));
-    menuBar->addMenu(m_menuManagerInterface->GetMenu(HelpMenuIdentifier));
+    m_menuManagerInterface->AddMenuToMenuBar(EditorMainWindowMenuBarIdentifier, FileMenuIdentifier, 100);
+    m_menuManagerInterface->AddMenuToMenuBar(EditorMainWindowMenuBarIdentifier, EditMenuIdentifier, 200);
+    m_menuManagerInterface->AddMenuToMenuBar(EditorMainWindowMenuBarIdentifier, GameMenuIdentifier, 300);
+    m_menuManagerInterface->AddMenuToMenuBar(EditorMainWindowMenuBarIdentifier, ToolsMenuIdentifier, 400);
+    m_menuManagerInterface->AddMenuToMenuBar(EditorMainWindowMenuBarIdentifier, ViewMenuIdentifier, 500);
+    m_menuManagerInterface->AddMenuToMenuBar(EditorMainWindowMenuBarIdentifier, HelpMenuIdentifier, 600);
```

Comment:
question: What are the numerical values here for? Could potentially be worth making a strong type to make this more obvious to the caller (see ManipulatorId and ManipulatorManagerId for an example, as well as what White Box does with VertexHandles, EdgeHandles etc....)

Reply:
Numerical values are sort keys, so it's just the ordering those menu items should have in the menu.
They're going to be pretty standard in all `Add` calls in the ActionManager, so I personally don't feel the need to weigh down the API by adding "AtPosition" to the function name, or enforce a strong type... but I'm open to discussion if you feel like this would increase readability :)